pkg/operator/deploymentcontroller/deployment_controller.go

```diff
@@ -151,42 +206,48 @@ func (c *DeploymentController) syncManaged(ctx context.Context, opSpec *opv1.Ope
 		return err
 	}
 
-	availableCondition := opv1.OperatorCondition{
-		Type:   c.name + opv1.OperatorStatusTypeAvailable,
-		Status: opv1.ConditionTrue,
-	}
-
-	if deployment.Status.AvailableReplicas > 0 {
-		availableCondition.Status = opv1.ConditionTrue
-	} else {
-		availableCondition.Status = opv1.ConditionFalse
-		availableCondition.Message = "Waiting for Deployment"
-		availableCondition.Reason = "Deploying"
-	}
-
-	progressingCondition := opv1.OperatorCondition{
-		Type:   c.name + opv1.OperatorStatusTypeProgressing,
-		Status: opv1.ConditionFalse,
+	updateStatusFuncs := []v1helpers.UpdateStatusFunc{
+		func(newStatus *opv1.OperatorStatus) error {
+			// TODO: set ObservedGeneration (the last stable generation change we dealt with)
+			resourcemerge.SetDeploymentGeneration(&newStatus.Generations, deployment)
+			return nil
+		},
 	}
 
-	if ok, msg := isProgressing(deployment); ok {
-		progressingCondition.Status = opv1.ConditionTrue
-		progressingCondition.Message = msg
-		progressingCondition.Reason = "Deploying"
+	// Set Available Condition
+	if slices.Contains(c.conditions, opv1.OperatorStatusTypeAvailable) {
+		availableCondition := opv1.OperatorCondition{
+			Type:   c.name + opv1.OperatorStatusTypeAvailable,
+			Status: opv1.ConditionTrue,
+		}
+		if deployment.Status.AvailableReplicas > 0 {
+			availableCondition.Status = opv1.ConditionTrue
+		} else {
+			availableCondition.Status = opv1.ConditionFalse
+			availableCondition.Message = "Waiting for Deployment"
+			availableCondition.Reason = "Deploying"
+		}
+		updateStatusFuncs = append(updateStatusFuncs, v1helpers.UpdateConditionFn(availableCondition))
 	}
 
-	updateStatusFn := func(newStatus *opv1.OperatorStatus) error {
-		// TODO: set ObservedGeneration (the last stable generation change we dealt with)
-		resourcemerge.SetDeploymentGeneration(&newStatus.Generations, deployment)
-		return nil
+	// Set Progressing Condition
+	if slices.Contains(c.conditions, opv1.OperatorStatusTypeProgressing) {
```

Comment:
old code always  had these, why make them "sometimes"?

Reply:
The idea was to have the possibility to make every condition optional. `NewDeploymentController` still always has this by default.